isisd/isis_spf_delay.c

```diff
@@ -0,0 +1,432 @@
+										  area->spf_delay_ietf[level-1]->holddown);
+      THREAD_TIMER_MSEC_ON (master, area->spf_delay_ietf[level-1]->family[family]->t_holddown, 
+      isis_spf_delay_event_holddown_expires,reinit,area->spf_delay_ietf[level-1]->holddown);
+  
+  } else if (area->spf_delay_ietf[level-1]->family[family]->state == ISIS_SPF_DELAY_STATE_SHORT_WAIT) {
+    // SHORT WAIT state behavior
+    
+    current_wait_time = area->spf_delay_ietf[level-1]->short_delay;
+
+    // We disable Holddown timer
+    if (isis->debugs & DEBUG_SPF_IETF)
+      zlog_debug ("ISIS-Spf (%s) L%d F%d IETF SPF delay : disabling holddown (thread %p)",
+                  area->area_tag,
+                  level,
+		  family,
+		  area->spf_delay_ietf[level-1]->family[family]->t_holddown);
+
+      THREAD_TIMER_OFF(area->spf_delay_ietf[level-1]->family[family]->t_holddown);
+  
+      // Start Holddown timer
+      if (isis->debugs & DEBUG_SPF_IETF)
+        zlog_debug ("ISIS-Spf (%s) L%d F%d IETF SPF delay launch HOLDDOWN timer\n",
+                    area->area_tag,
+		    level,
+		    family);
+      area->spf_delay_ietf[level-1]->family[family]->resettime = timeval_add_msec(time_now,
+										  area->spf_delay_ietf[level-1]->holddown);
+      THREAD_TIMER_MSEC_ON (master,
+			    area->spf_delay_ietf[level-1]->family[family]->t_holddown,
+                            isis_spf_delay_event_holddown_expires,
+			    reinit,
+			    area->spf_delay_ietf[level-1]->holddown);
+  } else {
+    // LONG WAIT state behavior
+    
+    current_wait_time = area->spf_delay_ietf[level-1]->long_delay;
+    // We disable Holddown timer
+    if (isis->debugs & DEBUG_SPF_IETF)
+      zlog_debug ("ISIS-Spf (%s) L%d F%d IETF SPF delay : disabling holddown (thread %p)",
+                  area->area_tag,
+		  level,
+		  family,
+		  area->spf_delay_ietf[level-1]->family[family]->t_holddown);
+
+    THREAD_TIMER_OFF(area->spf_delay_ietf[level-1]->family[family]->t_holddown);
+	 
+    // Start Holddown timer
+    if (isis->debugs & DEBUG_SPF_IETF)
+      zlog_debug ("ISIS-Spf (%s) L%d F%d IETF SPF delay launch HOLDDOWN timer\n",
+                  area->area_tag,
+		  level,
+		  family);
+    area->spf_delay_ietf[level-1]->family[family]->resettime = timeval_add_msec(time_now,
+										area->spf_delay_ietf[level-1]->holddown);
+    THREAD_TIMER_MSEC_ON (master,
+			  area->spf_delay_ietf[level-1]->family[family]->t_holddown,
+			  isis_spf_delay_event_holddown_expires,
+			  reinit,
+			  area->spf_delay_ietf[level-1]->holddown);
+  }
+
+  // We check if SPF is already scheduled
+  if (area->spf_delay_ietf[level-1]->family[family]->pending) {
+    if  (isis->debugs & DEBUG_SPF_IETF) {	
+      zlog_debug ("ISIS-Spf (%s) L%d F%d IETF SPF already scheduled, due in %ld msec",
+                  area->area_tag,
+		  level,
+		  family,
+		  timeval_elapsed(area->spf_delay_ietf[level-1]->family[family]->next_spf, 
+                                  time_now) / 1000);	
```

Comment:
Right, will fix it by using the thread_timer_remain. I was not aware of this function. Thanks for pointing it. I will check if next_spf is still necessary or not. If not, I will remove it.